src/identity/ed25519/ed25519-key-pair-identity.ts

```diff
@@ -30,7 +30,15 @@ export class Ed25519KeyPairIdentity extends PublicKeyIdentity {
       throw new Error("Invalid Mnemonic")
     }
     const seed = bip39.mnemonicToSeedSync(sanitized).slice(0, 32)
-    const keys = ed25519.generateKeyPair({ seed })
+    return Ed25519KeyPairIdentity.fromSeed(seed)
+  }
+
+  static fromSeed(seed: string | pki.ed25519.NativeBuffer) {
+    let _seed = seed?.slice(0, 32)
+    if (_seed.length !== 32) {
+      throw new Error("seed should have length of 32")
+    }
+    const keys = ed25519.generateKeyPair({ seed: _seed })
```

Comment:
`ed25519.generateKeyPair` will throw errors if `seed` isn't 32 characters or if it's the wrong type (specifically, not coercible into an `Uint8Array`). Should we just catch here instead of having our own validation?

Reply:
good point.  yeah let's remove validation and catch here instead